fs2/prediction_writing_callback.py

```diff
@@ -362,21 +366,15 @@ def synthesize_audio(self, outputs):
                 f"{self.output_key} does not exist in the output of your model"
             )
 
-        # wavs: [B (batch_size), T (samples)]
+        # wavs: [B (batch_size), C (channels), T (samples)]
         assert (
-            wavs.ndim == 2
-        ), f"The generated audio contained more than 2 dimensions. First dimension should be B(atch) and the second dimension should be T(ime) in samples. Got {wavs.shape} instead."
-        assert "output" in outputs and outputs["output"] is not None
-        assert wavs.shape[0] == outputs["output"].size(
+            wavs.ndim == 3
+        ), f"The generated audio did not contain 3 dimensions. First dimension should be B(atch) and the second dimension should be C(hannels) and third dimension should be T(ime) in samples. Got {wavs.shape} instead."
```

Comment:
Can this happen due to a user error (like providing the wrong kind of input file), or is this strictly due to a programmer error? If the latter, OK, if the former, I don't like using assert.

Reply:
It could happen due to a user error. What error would you prefer?